rocksdb_admin/admin_handler.cpp

```diff
@@ -377,6 +393,24 @@ bool AdminHandler::writeMetaData(
   return s.ok();
 }
 
+bool DeserializeKafkaPayload(
+    const string& payload,
+    KafkaOperationCode& op_code,
+    rocksdb::Slice& value) {
+  KafkaMessagePayload msg_payload;
+
+  if (DecodeThriftStruct<KafkaMessagePayload>(payload, &msg_payload)) {
+    op_code = msg_payload.op_code;
+    if (msg_payload.__isset.value) {
+      value = rocksdb::Slice(msg_payload.value);
```

Comment:
msg_payload.value is on the stack of function DeserializeKafkaPayload, and value internally keeps a reference to it. After returning from DeserializeKafkaPayload(), the content referenced by value will be invalid.

Reply:
Good point. Not sure why it didn't cause a seg fault when I tested.